internal/app/osd/internal/reg/reg.go

```diff
@@ -367,3 +370,23 @@ func (r *Registrator) Version(ctx context.Context, in *empty.Empty) (data *proto
 
 	return data, err
 }
+
+// Top implements the proto.OSDServer interface
+func (r *Registrator) Top(ctx context.Context, in *empty.Empty) (reply *proto.TopReply, err error) {
+	var procs []proc.ProcessList
+	procs, err = proc.List()
+	if err != nil {
+		return
+	}
+
+	var plist bytes.Buffer
+	enc := gob.NewEncoder(&plist)
```

Comment:
I wasn't sure the best way to implement this; I want to return back a `[]proc.ProcessList` but I didn't want to redefine it in the protobuf ( seems like a bad approach to dupe? ). So I decided to just return back a `[]byte` representation of the `[]proc.ProcessList` to work around this.

Reply:
I've never used `gob` before, but should be nice to use unless we want to allow non-osctl clients here, and `gob` is Go-specific imho? 

I have nothing against it, but at the same time `msgpack` might provide better interoperability with other clients in the future. I believe msgpack should be almost as efficient in this case.

Reply:
I'm not so familiar with protobuf, but I think we can let protobuf generate that `ProcessList` structure from definition and we can extend it the way we want? this way we can keep API to be purely protobuf-based?

Reply:
If we can keep it pure protobuf that has my vote, but I’m not sure on how to extend it.